### PR TITLE
Export HttpError and SpecOptions to Noms JS API

### PR DESCRIPTION
**Pull request by kalman**
I renamed HTTPError to HttpError for consistency with HttpBatchStore
(and XMLHttpRequest).

### CODE REVIEW

File: js/noms/src/noms.js [hunk 1]
```diff
@@ -59,7 +60,7 @@ export {
   getTypeOfValue,
 } from './type.js';
 export {equals, less} from './compare.js';
-export {default as Spec} from './spec.js';
+export {default as Spec, SpecOptions} from './spec.js';
```
Comment (arv) [inline js/noms/src/noms.js]: Inc minor version for this?

Comment (kalman) [inline js/noms/src/noms.js]: I did in a 2nd patchset (pushed with -f), did you see that?

Comment (arv) [inline js/noms/src/noms.js]: See it now.

**Comment by arv** [comment]
Still LGTM

**Pull request merged**